src/tests/dbus-tests/test_bcache.py

```diff
@@ -0,0 +1,132 @@
+                                           dbus_interface=self.iface_prefix + '.Manager.Bcache')
+        time.sleep(1)
+        self.addCleanup(self._force_remove, bcache_name, self.vdevs[0], self.vdevs[1])
+
+        bcache = self.get_object('/block_devices/' + bcache_name)
+        self.assertIsNotNone(bcache)
+
+        self.assertTrue(os.path.exists('/sys/block/%s/bcache' % bcache_name))
+
+        # check properties
+        sys_mode = self._get_mode(bcache_name)
+        dbus_mode = self.get_property(bcache, '.Block.Bcache', 'mode')
+        dbus_mode.assertEqual(sys_mode)
+
+        sys_state = self.read_file('/sys/block/%s/bcache/state' % bcache_name).strip()
+        dbus_state = self.get_property(bcache, '.Block.Bcache', 'state')
+        dbus_state.assertEqual(sys_state)
+
+        sys_block = self.read_file('/sys/block/%s/bcache/cache/block_size' % bcache_name)
+        dbus_block = self.get_property(bcache, '.Block.Bcache', 'block_size')
+        dbus_block.assertEqual(int(sys_block))
+
+        sys_size = self._get_size(bcache_name)
+        dbus_size = self.get_property(bcache, '.Block.Bcache', 'cache_size')
+        dbus_size.assertEqual(sys_size * BLOCK_SIZE)
+
+        sys_hits = self.read_file('/sys/block/%s/bcache/cache/stats_total' \
+                                  '/cache_hits' % bcache_name)
+        dbus_hits = self.get_property(bcache, '.Block.Bcache', 'hits')
+        dbus_hits.assertEqual(int(sys_hits))
+
+        sys_misses = self.read_file('/sys/block/%s/bcache/cache/stats_total' \
+                                    '/cache_misses' % bcache_name)
+        dbus_misses = self.get_property(bcache, '.Block.Bcache', 'misses')
+        dbus_misses.assertEqual(int(sys_misses))
+
+        sys_byhits = self.read_file('/sys/block/%s/bcache/cache/stats_total' \
+                                    '/cache_bypass_hits' % bcache_name)
+        dbus_byhits = self.get_property(bcache, '.Block.Bcache', 'bypass_hits')
+        dbus_byhits.assertEqual(int(sys_byhits))
+
+        sys_bymisses = self.read_file('/sys/block/%s/bcache/cache/stats_total' \
+                                      '/cache_bypass_misses' % bcache_name)
+        dbus_bymisses = self.get_property(bcache, '.Block.Bcache', 'bypass_misses')
+        dbus_bymisses.assertEqual(int(sys_bymisses))
+
+        # destroy the cache
+        bcache.BcacheDestroy(self.no_options, dbus_interface=self.iface_prefix + '.Block.Bcache')
+        time.sleep(1)
+
+        # make sure the bcache device is not on dbus
+        udisks = self.get_object('')
+        objects = udisks.GetManagedObjects(dbus_interface='org.freedesktop.DBus.ObjectManager')
+        self.assertNotIn(str(bcache.object_path), objects.keys())
+
+        self.assertFalse(os.path.exists('/dev/%s' % bcache_name))
+
+    def test_set_mode(self):
+        '''Test if it's possible to change cache mode on existing bcache'''
+
```

Comment:
And the same here or rename the tests to make sure this one runs after the create_destroy one.